lib/db/files_db.dart

```diff
@@ -54,21 +60,21 @@ class FilesDB {
   _initDatabase() async {
     Directory documentsDirectory = await getApplicationDocumentsDirectory();
     String path = join(documentsDirectory.path, _databaseName);
-    return await openDatabase(path,
-        version: _databaseVersion, onCreate: _onCreate);
+    return await openDatabaseWithMigration(path, dbConfig);
   }
 
   // SQL code to create the database table
-  Future _onCreate(Database db, int version) async {
-    await db.execute('''
-          CREATE TABLE $table (
+  static List<String> onCreate(String tablename) {
+    return [
+      '''
+          CREATE TABLE $tablename (
             $columnGeneratedID INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
             $columnLocalID TEXT,
             $columnUploadedFileID INTEGER,
             $columnOwnerID INTEGER,
             $columnCollectionID INTEGER,
             $columnTitle TEXT NOT NULL,
-            $columnDeviceFolder TEXT NOT NULL,
+            $columnDeviceFolder TEXT,
```

Comment:
Do you want to keep the `NOT NULL` constraint so that the `alterDeviceFolderToAllowNULL` makes sense?